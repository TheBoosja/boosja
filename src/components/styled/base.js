import styled from 'styled-components';
import { Link } from 'react-router-dom';

const Anchor = styled(Link)`
	color: ${({ theme }) => theme.textColor};
	background-color: ${({ theme }) => theme.primaryColor};
	padding: 1rem 2rem;
	text-decoration: none;
`;
Anchor.displayName = 'Anchor';

const Button = styled.button`
	background-color: ${({ theme }) => theme.primaryColor};
	border: none;
	color: ${({ theme }) => theme.textColor};
	font-family: inherit;
	font-size: ${({ large }) => (large ? '2rem' : '1.5rem')};
	padding: 1rem 2rem;
	cursor: pointer;
	transition: all 0.2s;

	&:hover {
		background-color: ${({ theme }) => theme.primaryColorLight};
	}
`;
Button.displayName = 'Button';

const Image = styled.img`
	display: block;
	height: 100%;
	object-fit: cover;
	width: 100%;
`;
Image.displayName = 'Image';

const Form = styled.form`
	display: grid;
	grid-template-columns: 30vw;
	align-content: center;
	justify-content: center;
`;
Form.displayName = 'Form';

const Input = styled.input`
	background-color: ${({ theme }) => theme.bgLight};
	border: none;
	color: inherit;
	font-family: inherit;
	font-size: 1.5rem;
	padding: 1rem 2rem;
	width: 100%;

	&:focus {
		outline: none;
	}
`;
Input.displayName = 'Input';

const Error = styled.label`
	display: block;
	color: red;
	font-size: 1.2rem;
	margin: 1rem 2rem;
`;
Error.displayName = 'Error';

export { Anchor, Button, Image, Form, Input, Error };
