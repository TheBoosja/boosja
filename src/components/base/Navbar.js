import React from 'react';
import _ from 'lodash';
import { connect } from 'react-redux';
import { Nav, NavItem } from '../styled';
import { withRouter } from 'react-router-dom';

export const Navbar = ({ isAuthenticated }) => {
	const navItems = [
		{ to: '/', label: 'Home', exact: true },
		{ to: '/profile', label: 'Profile', auth: true },
		{ to: '/search', label: 'Search' },
		{ to: '/signout', label: 'Sign Out', auth: true },
		{ to: '/auth', label: 'Sign In', auth: false }
	];

	const computedList = (auth) => {
		const filteredItems = _.filter(navItems, 
			item => _.isUndefined(item.auth) ? true : item.auth === auth);

		return _.map(filteredItems, ({ exact, to, label }, i) => 
			<NavItem exact={exact || false} to={to} key={i}>{label}</NavItem>);
	};

	return (
		<Nav>
			{computedList(isAuthenticated)}
		</Nav>
	);
};

function mapStateToProps({ auth: { isAuthenticated } }) {
	return { isAuthenticated };
}

export default withRouter(connect(mapStateToProps)(Navbar));