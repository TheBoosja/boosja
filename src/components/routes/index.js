import React from 'react';
import { Switch, Route } from 'react-router-dom';

import Home from '../Home';
import AuthPage from '../auth';

const Routes = () => {
	return (
		<Switch>
			<Route exact path='/' component={Home} />
			<Route exact path='/auth' component={AuthPage} />
		</Switch>
	)
}

export default Routes;