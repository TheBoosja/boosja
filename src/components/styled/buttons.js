import styled from 'styled-components';
import { Link } from 'react-router-dom';

export const Anchor = styled(Link)`
	color: ${({ theme }) => theme.textColor};
	background-color: ${({ theme }) => theme.primaryColor};
	padding: 1rem 2rem;
	text-decoration: none;
`;
Anchor.displayName = 'Anchor';

export const Button = styled.button`
	background-color: ${({ theme }) => theme.primaryColor};
	border: none;
	color: ${({ theme }) => theme.textColor};
	font-family: inherit;
	font-size: ${({ large }) => (large ? '2rem' : '1.5rem')};
	padding: 1rem 2rem;
	cursor: pointer;
	transition: all 0.2s;

	&:hover {
		background-color: ${({ theme }) => theme.primaryColorLight};
	}
`;
Button.displayName = 'Button';
