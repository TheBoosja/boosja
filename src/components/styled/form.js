import styled from 'styled-components';

export const Form = styled.form`
	display: grid;
	grid-template-columns: 50vw;
	align-content: center;
	justify-content: center;
`;
Form.displayName = 'Form';

export const Input = styled.input`
	background-color: ${({ theme }) => theme.bgLight};
	border: none;
	color: inherit;
	font-family: inherit;
	font-size: 2rem;
	padding: 1rem 2rem;
	width: 100%;

	&:focus {
		outline: none;
	}
`;
Input.displayName = 'Input';

export const Error = styled.span`
	display: block;
	color: red;
	font-size: 1.2rem;
	margin: 1rem 2rem;
`;
Error.displayName = 'Error';