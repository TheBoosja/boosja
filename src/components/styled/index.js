export { Title } from './typography';
export { Nav, NavItem } from './navbar';
export { Anchor, Button } from './buttons';
export { Form, Input, Error } from './form';