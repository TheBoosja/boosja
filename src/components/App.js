import React from 'react';
import styled, { ThemeProvider } from 'styled-components';
import theme from './theme';
import Navbar from './base/Navbar';
import Routes from './routes';

const Layout = styled.div`
	background-color: ${({ theme }) => theme.bgDark};
	display: grid;
	grid-template-columns: 10vw 1fr;
`;

const App = () => {
	return (
		<ThemeProvider theme={theme}>
			<Layout>
				<Navbar />

				<Routes />
			</Layout>
		</ThemeProvider>
	);
};

export default App;