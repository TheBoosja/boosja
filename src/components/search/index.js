import React from 'react';
import { Route } from 'react-router-dom';
import { SearchPage } from 'components/styled';

import SearchBar from './SearchBar';
import SearchList from './SearchList';

const Search = ({ match, history }) => {
	return (
		<SearchPage>
			<SearchBar {...{ match, history }} />
			<Route path={`${match.url}/:query/:page?`} component={SearchList} />
		</SearchPage>
	);
};

export default Search;
