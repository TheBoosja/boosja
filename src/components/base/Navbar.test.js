import React from 'react';
import { shallow } from 'enzyme';
import { Nav, NavItem } from 'components/styled';
import { Navbar } from './Navbar';

describe('Navbar', () => {
	const navbar = shallow(<Navbar />);

	it('renders properly', () => {
		expect(navbar).toMatchSnapshot();
	});

	it('contains a link to the homepage', () => {
		expect(navbar.find({ to: '/' }).exists()).toBe(true);
	});

	it('contains a link to the Search page', () => {
		expect(navbar.find({ to: '/search' }).exists()).toBe(true);
	});

	describe('when not authenticated', () => {
		const props = { isAuthenticated: false };
		const navbar = shallow(<Navbar {...props} />);

		it('contains a link to the Sign In page', () => {
			expect(navbar.find({ to: '/signin' }).exists()).toBe(true);
		});

		it('contains a link to the Register page', () => {
			expect(navbar.find({ to: '/register' }).exists()).toBe(true);
		});

		it('does not contain a link to the Profile', () => {
			expect(navbar.find({ to: '/profile' }).exists()).toBe(false);
		});

		it('does not contain a link to Sign out', () => {
			expect(navbar.find({ to: '/signout' }).exists()).toBe(false);
		});
	});

	describe('when authenticated', () => {
		const props = { isAuthenticated: true };
		const navbar = shallow(<Navbar {...props} />);

		it('contains a link to the Profile', () => {
			expect(navbar.find({ to: '/profile' }).exists()).toBe(true);
		});

		it('contains a link to Sign Out', () => {
			expect(navbar.find({ to: '/signout' }).exists()).toBe(true);
		});

		it('does not contain a link to the Sign In page', () => {
			expect(navbar.find({ to: '/signin' }).exists()).toBe(false);
		});

		it('does not contain a link to the Register page', () => {
			expect(navbar.find({ to: '/register' }).exists()).toBe(false);
		});
	});

	describe('styled', () => {
		it('Nav renders properly', () => {
			const nav = shallow(<Nav />);
			expect(nav).toMatchSnapshot();
		});

		it('NavItem renders properly', () => {
			const navItem = shallow(<NavItem />);
			expect(navItem).toMatchSnapshot();
		});
	});
});
