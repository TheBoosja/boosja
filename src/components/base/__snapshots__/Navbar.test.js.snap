// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <NavItem
          exact={true}
          to="/"
>
          Home
</NavItem>,
        <NavItem
          exact={false}
          to="/search"
>
          Search
</NavItem>,
        <NavItem
          exact={false}
          to="/signin"
>
          Sign In
</NavItem>,
        <NavItem
          exact={false}
          to="/register"
>
          Register
</NavItem>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "children": "Home",
          "exact": true,
          "to": "/",
        },
        "ref": null,
        "rendered": "Home",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "children": "Search",
          "exact": false,
          "to": "/search",
        },
        "ref": null,
        "rendered": "Search",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "children": "Sign In",
          "exact": false,
          "to": "/signin",
        },
        "ref": null,
        "rendered": "Sign In",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "function",
        "props": Object {
          "children": "Register",
          "exact": false,
          "to": "/register",
        },
        "ref": null,
        "rendered": "Register",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <NavItem
            exact={true}
            to="/"
>
            Home
</NavItem>,
          <NavItem
            exact={false}
            to="/search"
>
            Search
</NavItem>,
          <NavItem
            exact={false}
            to="/signin"
>
            Sign In
</NavItem>,
          <NavItem
            exact={false}
            to="/register"
>
            Register
</NavItem>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "children": "Home",
            "exact": true,
            "to": "/",
          },
          "ref": null,
          "rendered": "Home",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "children": "Search",
            "exact": false,
            "to": "/search",
          },
          "ref": null,
          "rendered": "Search",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "children": "Sign In",
            "exact": false,
            "to": "/signin",
          },
          "ref": null,
          "rendered": "Sign In",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "children": "Register",
            "exact": false,
            "to": "/register",
          },
          "ref": null,
          "rendered": "Register",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Navbar styled Nav renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <NavigationList />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <NavigationList />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Navbar styled NavItem renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavItem />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <NavigationListItemLink />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <NavigationListItemLink />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
